lib/pure/stats.nim

```diff
@@ -309,104 +290,102 @@ proc partition[T](A: var openArray[T], left, right, K : Natural,
+  elif K < pInd: # if K is less than the pivot index
+    result= quickSelect(A, left, pInd - 1, K, myCmp)
 
   # if K is greater than the pivot index
   else:
-    return quickSelect(A, pInd + 1, right, K, myCmp)
+    result= quickSelect(A, pInd + 1, right, K, myCmp)
 
+type MedianMode* = enum
+  mdlow, mdhigh
 
-proc median_low*[T](A: var openArray[T], left, right : Natural, 
-                         myCmp : proc(x,y:T):int {.nimcall} = cmp[T]): T =
+proc median*[T](A: var openArray[T], left, right: Natural, Mode:MedianMode,
+                    myCmp: proc(x, y: T): int {.nimcall.} = cmp[T]): T =
   ##[
   The **median M** of an array/sequence of comparable items is defined such that 
   
   in case the array/sequence has *odd* length ``(right-left+1)`` : 
-     ``(m-1)/2`` elements are lower or equal and ``(m-1)/2`` elements are higher or equal than **M**
+     ``(m-1)/2`` elements are lower or equal and 
+     ``(m-1)/2`` elements are higher or equal than **M** .
 
-     In this case  ``median_low(A) == median_high(A)``
+     In this case  ``Mode == mdlow`` and ``Mode == mdhigh`` are equal.
 
   in case the array/sequence has *even* length the median is only defined for
      arrays of ``SomeNumber`` elements - see below -
 
-     In general **median_low(A)** has the property that ``m/2`` elements are less than or equal this value
-     and ``m/2+1`` elements are greater than or equal this value.
+     In general **median Mode:mdlow** has the property that 
+     ``m/2``   elements are less than or equal this value  and
+     ``m/2+1`` elements are greater than or equal this value.
 
-     **median_high(A)** has the property that ``m/2`` elements are greater or equal than this value and
+     **median Mode:mdhigh** has the property that 
+     ``m/2``   elements are greater or equal than this value and
      ``m/2+1`` elements are less or equal than this value.
 
   >>> **A** will be modified
   ]##
-  quickSelect(A, left,right,(right+left) div 2, myCmp)
 
+  if  Mode == mdlow :
+    quickSelect(A, left, right, (right+left) div 2, myCmp)
+  else :
+    quickSelect(A, left, right, (right+left+1) div 2, myCmp)
 
-proc median_high*[T](A: var openArray[T], left, right : Natural, 
-                         myCmp : proc(x,y:T):int {.nimcall} = cmp[T]): T =
-  quickSelect(A, left,right,(right+left+1) div 2, myCmp)
 
-proc median*[T:SomeNumber](A: var openArray[T], left, right : Natural,
-                    myCmp : proc(x,y:T):int {.nimcall} = cmp[T]): float =
-  ##[This is a special version which returns a float value. 
-    If the length (right+1-left) is *even*, it returns (median_low + median_high)*0.5
+proc median*[T: SomeNumber](A: var openArray[T], left, right: Natural,
+                    myCmp: proc(x, y: T): int {.nimcall.} = cmp[T]): float =
+  ##[This is a special version which returns a float value.
+    If the length (right+1-left) is *even*,
+    it returns (medianLow + medianHigh)*0.5
 
   >>> **A** will be modified
```

Comment:
use admonition which renders better:
```suggestion
  ## .. warning:: `A` will be modified.
```
(also, can you make it more precise? eg `A` will be modified if unsorted?)